mlxtend/frequent_patterns/fpcommon.py

```diff
@@ -163,6 +164,19 @@ def valid_input_check(df, null_values=False):
             "Please use a DataFrame with bool type",
             DeprecationWarning,
         )
+
+        # If null_values is True but no NaNs are found, raise an error
```

Comment:
Just curious, would this lead to inaccurate metrics or is it just inefficient. Or are there other reasons why this would not be permitted?

Reply:
No, this one is just inefficient. Normally if I don't have an array with null values but null_values is True, it'd generate the final dataframe with metrics when null values do not exist (as original approach). The only reason I included is to remain consistent and still raise the error if it is the case. But since null_values is set False as default, maybe there's no point of including this condition as the user may not intend to set null_values to True if there are no NaNs in their array. Up to you, if you want I can completely exclude it.

Reply:
Thanks, that's what I thought! Sounds good! In this case, maybe we should not raise an error but a warning that this may be inefficient, and that `null_values` can be set to False for a speed-up. What do you think?

Reply:
Fixed! Please look at the latest commit.